sdk/lib/core/uri.dart

```diff
@@ -512,21 +554,38 @@ abstract interface class Uri {
 
   /// The fragment identifier component.
   ///
+  /// A fragment identifier is a string after URI, after the hash, which
+  /// identifies something specific as a function of the document. For
+  /// a user interface Web document such as HTML page, it traditionally
+  /// identifies a section or part of the page. In modern pages, it can
+  /// indicate a particular view of the page.
+  ///
   /// The value is the empty string if there is no fragment identifier
   /// component.
   String get fragment;
 
-  /// The URI path split into its segments.
+  /// Same as the URI [path], but split into its segments.
+  ///
+  /// This is an alternative way to access the [path] component and
+  /// is useful for when a program needs to access components of a URI
+  /// such as the immediate parent folder of the current file.
+  /// This kind of field is valuable for extracting folder names
+  /// for display in a GUI's breadcrumb navigation component.
+  ///
+  /// For example, if iterating through files in several local folders 
+  /// that have a similar folder and file naming scheme, you could modify
+  /// `pathSegments` to hit every file that matches a pattern within every
+  /// folder that matches a pattern.
```

Comment:
I think that paragraph is too vague to be useful.

Giving lots of examples of ways to *use* a value means attempting to guess the user's needs.
If we guess wrong, it's just noise to the reader, useless examples for something they're not doing.

Each example should show *something* generally useful